tests/test_picking_batch.py

```diff
@@ -305,3 +307,175 @@ def test14_is_valid_location_dest_failure_unknown_location(self):
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.update_picking(package_name=self.package_one.name)
+        move_line_id = picking.move_line_ids[0].id
+        reason = 'missing item'
+
+        # It doesn't actually matter what the picking type is for this test.
+        # The important thing is that the new picking is of type picking_type
+        picking_type = self.picking_type_pick
+        batch.unpickable_item(move_line_id=move_line_id,
+                              reason=reason,
+                              picking_type_id=picking_type.id)
+        new_picking_type = self.package_one.move_line_ids.picking_id.picking_type_id  # noqa
+
+        self.assertEqual(picking.state, 'cancel')
+        self.assertEqual(batch.state, 'done')
+        self.assertEqual(new_picking_type.name, picking_type.name)
+
+    def test17_unpickable_item_move_line_not_found(self):
+        """
+        Tests that a ValidationError is raised if the move_line_id cannot be
+        found
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.state = 'done'
+        reason = 'missing item'
+
+        with self.assertRaises(ValidationError) as err:
+            batch.unpickable_item(move_line_id=999,
+                                  reason=reason,
+                                  picking_type_id=None)
+            self.assertEqual(err.exception.name,
+                             'Cannot find the operation')
+
+    def test18_unpickable_item_wrong_batch(self):
+        """
+        Tests that a ValidationError is raised if the move_line_id is not on
+        the Batch that we requested.
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        # Create a quant and picking for a different package
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=self.package_two.id)
+        different_picking = self.create_picking(
+            self.picking_type_pick,
+            products_info=self.pack_4apples_info,
+            confirm=True,
+            assign=True)
+        different_picking.update_picking(package_name=self.package_two.name)
+        move_line_id = different_picking.move_line_ids[0].id
+        reason = 'missing item'
+        with self.assertRaises(ValidationError) as err:
+            batch.unpickable_item(move_line_id=move_line_id,
+                                  reason=reason,
+                                  picking_type_id=None)
+            self.assertEqual(err.exception.name,
+                             'Move line is not part of the batch')
+
+    def test19_unpickable_item_invalid_state_cancel(self):
+        """
+        Tests that a ValidationError is raised if the picking is on a state
+        of cancel
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.state = 'cancel'
```

Comment:
Again we cannot do this. In this test you can force_validate to have it in state done.

Reply:
As discussed, the only way to make this test pass is to manually set the state to "cancel".  Have added a comment in detailing the full reasons